Packet++/src/IPReassembly.cpp

```diff
@@ -342,7 +342,15 @@ Packet* IPReassembly::processPacket(Packet* fragment, ReassemblyStatus& status,
 			PCPP_LOG_DEBUG("[FragID=0x" << std::hex << fragWrapper->getFragmentId() << "] Got first fragment, allocating RawPacket");
 
 			// create the reassembled packet and copy the fragment data to it
-			fragData->data = new RawPacket(*(fragment->getRawPacket()));
+
+			// copy only data from the beginning of the fragment to the end of IP layer payload.
+			// Don't copy data beyond it such as packet trailer
+			auto fragmentRawPacket = fragment->getRawPacket();
+			auto rawDataLen = fragWrapper->getIPLayerPayload() - fragmentRawPacket ->getRawData() + fragWrapper->getIPLayerPayloadSize();
```

Comment:
Can't we create a getFragmentLength() ?

Reply:
The `fragment` variable here is actually an instance of type `Packet` which is the first IP fragment. Since it's a packet it doesn't make sense to add a method like `getFragmentLength()`